app_src/utils/image_operations.py

```
@@ -1,24 +1,92 @@
+        or 'ANDROID_ARGUMENT' in os.environ
+        or os.getenv("MAIN_ACTIVITY_HOST_CLASS_NAME")
+    )
 
-if on_android_platform():
-    PythonActivity = _LazyJavaClass("PythonActivity", "org.kivy.android.PythonActivity")
+def _on_pydroid_app():
+    return "ru.iiec.pydroid3" in os.environ.get("PYTHONHOME", "")
+
+# Local _LazyJavaClass — defers jnius import to first use, not module level
+class _LazyJavaClass:
+    __slots__ = ("_python_name", "_java_name")
```

Ruff (0.16.1)

[warning] `_LazyJavaClass.__slots__` is not sorted

Apply a natural sort to `_LazyJavaClass.__slots__`

(RUF023)